components/fxa-client/ios/FxAClient/FirefoxAccount.swift

```diff
@@ -154,6 +154,7 @@ open class FirefoxAccount {
                 fxa_bytebuffer_free(profileBuffer)
                 let profile = Profile(msg: msg)
                 DispatchQueue.main.async { completionHandler(profile, nil) }
+                self.tryPersistState()
```

Comment:
Is calling this after/racing with the completion handler correct?

Reply:
Yup, I believe it should be alright as this state is only for our use anyway.